scripts/cloudant/generate_api_key.py

```diff
@@ -11,9 +11,14 @@
     args = parser.parse_args()
 
     cloudant_instance = authenticate_cloudant_instance(args.username)
-    api_key, api_password = cloudant_instance.generate_api_key()
+    api_key, api_password = cloudant_instance.generate_api_key().ask_and_run()
     print 'New API key generated.'
     print 'API Key:', api_key
     print 'API Password:', api_password
+
+    ask_runs = []
+
     for database in args.databases:
-        cloudant_instance.get_db(database).grant_api_key_access(api_key)
+        ask_runs.append(cloudant_instance.get_db(database).grant_api_key_access(api_key))
```

Comment:
(same question about performing the action before asking, looks like there's a few other places where it seems like this happens too)